Pear/Controller/CommunityViewController.swift

```diff
@@ -143,17 +152,33 @@ extension CommunityViewController: UITableViewDataSource {
 
 // MARK: - SearchBarDelegate
 extension CommunityViewController: UISearchBarDelegate {
-
-    private func searchUsers(query: String) {
-        // TODO: Fill in with network call
-    }
-
-    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
-        if searchText.isEmpty {
-            getUsers()
-        } else {
+    
+    func getUsers() {
+        if let searchText = searchBar.text {
             searchUsers(query: searchText)
+        } else {
+            getAllUsers()
         }
     }
 
+    func searchUsers(query: String) {
+        NetworkManager.getSearchedUsers(searchText: query, completion: { [weak self] result in
+            guard let self = self else { return }
```

Comment:
What does this line do exactly

Reply:
The guard? Because you have `[weak self]`, `self `will be optional in the body of the closure. You use `weak` so that the closure doesn't retain a strong reference to `self`, preventing it from being deallocated.